src/DataMixin.js

```diff
@@ -39,12 +39,20 @@ module.exports = {
     this.setState(buildInitialState(nextProps));
   },
 
-  componentWillMount() {
-    // Do the initial sorting if specified.
-    var {sortBy, data} = this.state;
-    if (sortBy) {
-      this.setState({ data: sort(sortBy, data) });
+  componentWillReceiveProps(nextProps) {
+    //If update is false, then preserve the sortBy order, but update
+    //everything else
+    if (nextProps.update === false ){
```

Comment:
@poojareddyintuit  This is pretty clever, but I feel like that forces us (the component user) to have to manually manage the update prop. The table should take care of that for us - if sortBy was defined and the data changes the table should re-sort the data when re-rendering the table.